app/init/global_event_handler.js

```diff
@@ -225,8 +224,8 @@ class GlobalEventHandler {
 
         dispatch(setServerVersion(''));
         Client4.serverVersion = '';
-        PushNotifications.setApplicationIconBadgeNumber(0);
-        PushNotifications.cancelAllLocalNotifications(); // TODO: Only cancel the notification that belongs to this server
+
+        PushNotifications.clearNotifications();
```

Comment:
we can actually remove it from here, as if the user is logged in, this function will call `logout`